### PR TITLE
Retain execution db records on running keep-logs

### PR DESCRIPTION
**Pull request by lorenzgerber**
These changes establish the --keep-logs mode as a consistent state on which still proper 'clean' can be run. This state will allow to copy/zip the meta data.

### CODE REVIEW

File: src/main/groovy/nextflow/cli/CmdClean.groovy [hunk 1]
```diff
@@ -175,9 +175,16 @@ class CmdClean extends CmdBase implements CacheBase {
...
+        // used as trigger to proceed with delete. In
+        // case of keepLogs, proceed by default
+        def proceed
+        if (keepLogs){
+            proceed = true
+        } else {
+            proceed = currentCacheDb.removeTaskEntry(hash)
+        }
+        if( proceed ) {
             // delete folder
```
Comment (pditommaso) [inline src/main/groovy/nextflow/cli/CmdClean.groovy]: This change can be refactored as 

```
def deleted = keepLogs || currentCacheDb.removeTaskEntry(hash)
```

or 

```
def proceed = keepLogs || currentCacheDb.removeTaskEntry(hash)
```

File: src/main/groovy/nextflow/cli/CmdClean.groovy [hunk 2]
```diff
@@ -236,7 +243,7 @@ class CmdClean extends CmdBase implements CacheBase {
                     result = false
                     if(!quiet) System.err.println "Failed to remove $dir"
                 }
-                
+
                 result ? FileVisitResult.CONTINUE : FileVisitResult.TERMINATE
```
Comment (pditommaso) [inline src/main/groovy/nextflow/cli/CmdClean.groovy]: try to avoid to commit empty/unnecessary changes (I know a bit paranoid .. 😅)

**Comment by pditommaso** [comment]
OK, I've merged it manually. Thanks a lot for your contribution.

File: src/main/groovy/nextflow/cli/CmdClean.groovy (hunk 2 above)
Comment (lorenzgerber) [reply src/main/groovy/nextflow/cli/CmdClean.groovy]: nah, sorry, you're of course right. I appreciate your comments :)
